crates/kornia-imgproc/src/interpolation/bicubic.rs

```diff
@@ -0,0 +1,90 @@
+use kornia_image::{allocator::ImageAllocator, Image};
+
+pub(crate) fn bicubic_interpolation<const C: usize, A: ImageAllocator>(
+    image: &Image<f32, C, A>,
+    u: f32,
+    v: f32,
+    c: usize,
+) -> f32 {
+    let (rows, cols) = (image.rows(), image.cols());
+
+    let iu = u.trunc() as usize;
+    let iv = v.trunc() as usize;
+
+    let x = u - iu as f32;
+    let y = v - iv as f32;
+
+    let get = |row: usize, col: usize| -> f32 {
+        let row = row.clamp(0, rows - 1);
+        let col = col.clamp(0, cols - 1);
+
+        *image.get_unchecked([row, col, c])
+    };
+
+    let mut f = [[0.0; 2]; 2]; // pixel value
+    let mut fx = [[0.0; 2]; 2]; // deivative in x dir
+    let mut fy = [[0.0; 2]; 2]; // derivative in y dir
+    let mut fxy = [[0.0; 2]; 2]; // derivative in xy
+
+    // using central derivative approximative
+    for j in 0..2 {
+        let rr = iv + j as usize;
+        for i in 0..2 {
+            let cc = iu + i as usize;
+            f[j][i] = get(rr, cc);
+            fx[j][i] = (get(rr, cc + 1) - get(rr, cc - 1)) * 0.5;
```

Comment:
the `get` is an expensive op as you compute each time the linear index to the underlaying array storage. For continuous postions (in the x direction) you can just add or substract positions.